### ISSUE TITLE
Installing explicit package list file fails with "Packages missing in current osx-64 channels"

### ISSUE THREAD

**Issue by robintw** [conda/conda#2619]
I have a conda environment file that was created with the `--explicit` option, thus providing the URLs for all of the packages. It was created on OS X, but when I try to install it by running `conda create -n test --file conda-explicit-hotbar` I get an error saying: `Error:  Packages missing in current osx-64 channels:` followed by a list of all of the packages in the file.

The environment file is available at https://gist.github.com/d8e29769e65192b83527b1af01a02174, and I've tried downloading some of the URLs and they download absolutely fine.

Does anyone have any idea what is going on here? I archived my environment to this file assuming that I could easily re-create it (particularly as it has the URLs in it) - but I can't seem to.

**Comment by robintw** [comment]
In case it helps, the output of `conda info` is:

```
Using Anaconda Cloud api site https://api.anaconda.org
Current conda install:

             platform : osx-64
        conda version : 4.0.8
  conda-build version : 0+unknown
       python version : 3.5.1.final.0
     requests version : 2.9.1
     root environment : /Users/robin/anaconda3  (writable)
  default environment : /Users/robin/anaconda3
     envs directories : /Users/robin/anaconda3/envs
        package cache : /Users/robin/anaconda3/pkgs
         channel URLs : https://conda.anaconda.org/t/<TOKEN>/conda-forge/osx-64/
                        https://conda.anaconda.org/t/<TOKEN>/conda-forge/noarch/
                        https://conda.anaconda.org/t/<TOKEN>/robintw/osx-64/
                        https://conda.anaconda.org/t/<TOKEN>/robintw/noarch/
                        https://repo.continuum.io/pkgs/free/osx-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/osx-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /Users/robin/.condarc
    is foreign system : False
```

**Comment by mcg1969** [comment]
Are you sure this was the full output of the `--explicit` command? It should have some comments at the top, as well as a line with nothing but `@EXPLICIT` before the first URL.

So I manually added `@EXPLICIT` back in, and save the file into the filename `tmp`. Then I tried `conda create -n test --file tmp` with the latest build of conda. I get this:
```
Error: no package 'numba::numba-0.22.1-np110py27_13.tar.bz2' in index
```
And sure enough, that package is no longer there for some reason. But if I remove just that one file, it proceeds just fine. I tried 4.0.8, and while I didn't let it finish to completion, it didn't give me the errors you're reporting, either.
